builtin/logical/pki/storage.go

```diff
@@ -67,6 +67,67 @@ func (e keyEntry) isManagedPrivateKey() bool {
 	return e.PrivateKeyType == certutil.ManagedPrivateKey
 }
 
+type issuerUsage uint
+
+const (
+	ReadOnlyUsage   issuerUsage = iota
+	IssuanceUsage   issuerUsage = 1 << iota
+	CRLSigningUsage issuerUsage = 1 << iota
+
+	// When adding a new usage in the future, we'll need to create a usage
+	// mask field on the IssuerEntry and handle migrations to a newer mask,
+	// inferring a value for the new bits.
+	AllIssuerUsages issuerUsage = ReadOnlyUsage | IssuanceUsage | CRLSigningUsage
+)
+
+var namedIssuerUsages = map[string]issuerUsage{
+	"read only":            ReadOnlyUsage,
+	"issuing certificates": IssuanceUsage,
+	"CRL signing":          CRLSigningUsage,
+}
+
+func (i *issuerUsage) ToggleUsage(usage issuerUsage) *issuerUsage {
```

Comment:
It's a bit weird here that we are taking a pointer reference and returning the same pointer ref, it would lead me to think that I'm getting a new copy of the object, but I actually have two references to the same thing and I've modified the original one.

I think you did this to daisy chain the Toggles which we could more or less accomplish with the following api?

```
func (i *issuerUsage) ToggleUsage(usages ...issuerUsage) {
	for _, usage := range usages {
		*i ^= usage
	}
}
```

Reply:
Ah that's nicer, I'll do that.